Devscord.DiscordFramework/ControllersService.cs

```diff
@@ -38,11 +38,20 @@ public void Run(DiscordRequest request, Contexts contexts)
             using (LogContext.PushProperty("Contexts", contexts))
             {
                 var readAlwaysMethods = this._controllersContainer.WithReadAlways;
-                Task.Run(() => RunMethods(request, contexts, readAlwaysMethods, true));
+                var readAlwaysTask = Task.Run(() => RunMethods(request, contexts, readAlwaysMethods, true));
+                
+                Task commandsTask = null;
                 if (request.IsCommandForBot)
                 {
                     var discordCommandMethods = this._controllersContainer.WithDiscordCommand;
-                    Task.Run(() => RunMethods(request, contexts, discordCommandMethods, false));
+                    commandsTask = Task.Run(() => RunMethods(request, contexts, discordCommandMethods, false));
+                }
+
+                // ReadAlwaysMethods should be first in throwing exception, bcs every ReadAlways exception is Error
+                await readAlwaysTask;
+                if (commandsTask != null)
+                {
+                    await commandsTask;
                 }
             }
         }
```

Comment:
tu nie mam pełnego podglądu pliku
w linii 70 zmień jeszcze Parallel.ForEach(controllers, controllerInfo =>  
na zwykły foreach 
później zastąpimy to biblioteką która ogarnia asynchroniczność automatycznie pod spodem (dobieranie wszystkiego według opłacalności a nie naszego "wydaje mi sie że tu będzie lepiej")